File: libraries/js/Tweenings/circular.js
class Circular {
    static evaluate(tweeningBehaviour, t, b, c, d){
        switch(tweeningBehaviour) {
            case ETweeningBehaviour.IN:
                return Circular.in(t,b,c,d);
            case ETweeningBehaviour.OUT:
                return Circular.out(t,b,c,d);
            case ETweeningBehaviour.IN_OUT:
                return Circular.inOut(t,b,c,d);
            default:
                return null;
        }
    }
    

    static in(t, b, c, d){
        return -c * (Math.sqrt(1 - (t /= d) * t) - 1) + b;
    }


    static out(t, b, c, d){
        return c * Math.sqrt(1 - (t = t / d -1) * t) + b;
    }

    
    static inOut(t, b, c, d){
        if ((t /= d / 2) < 1){ 
            return -c / 2 * (Math.sqrt(1 - t * t) - 1) + b;
        }

		return c / 2 * (Math.sqrt(1 - (t -= 2) * t) + 1) + b;
    }
}

File: libraries/python/Tweenings/circular.py
from ETweeningBehaviour import ETweeningBehaviour as ETB
import math

class Circular:
    def evaluate(tweeningBehaviour, t, b, c, d) :
        if tweeningBehaviour == ETB.IN:
            return Circular.in_circular(t,b,c,d)
        elif tweeningBehaviour == ETB.OUT:
            return Circular.out(t,b,c,d)
        elif tweeningBehaviour == ETB.IN_OUT:
            return Circular.inOut(t,b,c,d)

        return None

    def in_circular(t, b, c, d):
        t = t / d
        return -c * (math.sqrt(1 - t * t) - 1) + b
    


    def out(t, b, c, d):
        t = t / d - 1
        return c * math.sqrt(1 - t * t) + b
    

    
    def inOut(t, b, c, d):
        t = t / (d / 2)
        if (t < 1):
            return -c / 2 * (math.sqrt(1 - t * t) - 1) + b
        
        t = t - 2
        return c / 2 * (math.sqrt(1 - t * t) + 1) + b
